trio/_core/_asyncgens.py

```diff
@@ -12,6 +16,15 @@
 # Used to log exceptions in async generator finalizers
 ASYNCGEN_LOGGER = logging.getLogger("trio.async_generator_errors")
 
+if TYPE_CHECKING:
```

Comment:
I don't understand why this branch is needed. Since you have `from __future__ import annotations`, I would expect that it works fine to use the fully-specified types in `x: Type` contexts and the base types in `x = Type()` contexts.

Reply:
never mind, I see this was already discussed